flaskProject1/app.py

```diff
@@ -256,11 +256,21 @@ def overall5():
 
 @app.route("/grades")
 def grades():
-    if session["email"]:
-        grade = session["grade"]
-        email = session["email"]
-        name = session['name']
-    return render_template("mainpages/grades.html",email = email, name = name, grade=grade)
+    try:
+        if session["email"]:
+            grade = session["grade"]
+            email = session["email"]
+            name = session['name']
+        return render_template("mainpages/grades.html", email=email, name=name, grade=grade)
+    except:
+        return redirect(url_for("logged_in"))
```

Comment:
Good implementation!

Reply:
> Very well done. Really happy that the test are working

You should fix DB lesson 6. The page isn't working!